### DISCUSSION TITLE
Wilms Tumor Dataset Annotation (SCPCP000006)

### DISCUSSION THREAD

**Discussion by maud-p** [AlexsLemonade/OpenScPCA-analysis#635] [Propose a new analysis]
### Proposed analysis

Wilms tumor (WT) is the most common pediatric kidney cancer characterized by an exacerbated intra- and inter- tumor heterogeneity. The genetic landscape of WT is very diverse in each of the histological contingents. The COG classifies WT patients into two groups: the favorable histology and diffuse anaplasia. Each of these groups is composed of the blastemal, epithelial, and stromal populations of cancer cells in different proportions, as well as cells from the normal kidney, mostly kidney epithelial cells, endothelial cells, immune cells and normal stromal cells (fibroblast). 

### Scientific goals

Here, we first aim to annotate the Wilms Tumor snRNA-seq samples in the SCPCP000006 (n=40) dataset. To do so we will:
•	Provide annotations of normal cells composing the kidney, including normal kidney epithelium, endothelium, stroma and immune cells
•	Provide annotations of tumor cell populations that may be present in the WT samples, including blastemal, epithelial, and stromal populations of cancer cells
Based on the provided annotation, we would like to additionally provide a reference of marker genes for the three cancer cell populations, which is so far lacking for the WT community. 

### Methods or approach

**Step 1 – acquisition of INPUT data**
To ensure a uniform analysis across modules, we will start the analysis with the processed objects available on the ScPCA Portal (_processed.rds). From my understanding, these objects have already undergone empty droplets removal, low quality cells filtering, and normalization and should be ready to go!

**Step 2 – dimensional reduction and clustering**
We hypothesize that after dimensional reduction, a cluster contains a relatively homogeneous population and will aim to annotate the dataset at the cluster level first. Refined annotation can be performed afterwards.  

**Step 3 – identification of immune and endothelium (normal) cells using annotation transfer from the healthy fetal kidney atlas and first felling into the cell composition**
Wilms tumor have been reported to be closer to the fetal kidney as mature kidney. We will first map the snRNA-Seq to the fetal kidney reference (from the kidney cell atlas [kidneycellatlas.org](https://www.kidneycellatlas.org/), [1]). This will allow the straight identification and annotation of immune and endothelial cells. Cells labelled as fetal nephron should contain (i) normal kidney epithelium cells, (ii) epithelial and blastemal cancer cells. Cells labelled as stromal cells might contain (i) normal fibroblasts, (ii) cancer associated fibroblasts (CAF) as well as (iii) stromal cancer cells.

**Step 4 – identification of malignant cells using inferCNV (and/or Numbat)**
We will use inferCNV to infer copy number alterations (immune and endothelial cells as normal cells for the reference). Ideally, we would like to compare results with other tools such as Numbat (if BAM file available) or copykat. Cells with CNV will be annotated as malignant, the others as normal cells. In non-obvious cases, cells will be annotated as the level of a Seurat cluster, as cells composing a cluster should be relatively homogeneous in cell type. 

**Step 5 – integration of CNV, label transfer and marker genes to complete the annotation**
We will finally integrate the information obtained in the previous steps to annotated as the following:
-	Cancer cells (CNV enriched clusters)
o	Epithelial cancer cells: labeled as fetal kidney epithelium or cap mesenchyme, epithelial markers PODXL, CDH1, LTL, WT1 (?)
o	Stromal cancer cells: labeled as stromal fetal cells + stromal markers VIM, collagenases
o	Blastemal cancer cells: labeled as cap mesenchyme mostly + blastemal marker CITED1, SIX2 (?)
-	Normal cells (CNV depleted clusters)
o	Immune cells
o	Endothelial cells
o	Normal kidney epithelium (podocyte PODXL+, renal vesicle, uteric bud)
o	Normal kidney stroma, mostly fibroblasts

**Step 6 – validation by integration of the 40 samples**
Next, we will integrate the 40 snRNA-Seq using scVI or harmony, perform dimensional reduction and clustering. This will allow to validate our annotations, as cells from the same cell type should cluster together. We will validate this by comparing compatible cluster marker expression across all integrated datasets. 

**Step 7 – identification of marker genes for each cell subtype using differential expression analysis**
Finally, we would like to provide the WT community with universal marker genes for a rapid identification of the different cell types found within the tumors. To do so, we will use pseudobulk differential expression analyses (DElegate package [2-3]) to find markers of the different cell types using the function FindAllMarkers2 (default parameters, patient as replicate). Additionally, we could compare relapse and non-relapse samples per cell type using the function findDE (replicate_column = "patient", method = “edger”) to evaluate if a specific phenotype within the cancer cells or the microenvironment could indicate relapse in WT.

### Existing modules

-

### Input data

To ensure a uniform analysis across modules, we will start the analysis with the processed objects available on the ScPCA Portal (_processed.rds). From my understanding, these objects have already undergone empty droplets removal, low quality cells filtering, and normalization and should be ready to go!

### Scientific literature

[1] [Spatiotemporal immune zonation of the human kidney | Science](https://www.science.org/doi/10.1126/science.aat5031)
[2] [Single-cell RNA-seq differential expression tests within a sample should use pseudo-bulk data of pseudo-replicates | bioRxiv](https://www.biorxiv.org/content/10.1101/2023.03.28.534443v1)
[3] [GitHub - cancerbits/DElegate: Wrapper and helper functions to use bulk RNA-seq differential expression methods with single-cell data](https://github.com/cancerbits/DElegate)

### Other details

**Suggestion**
Correct identification of normal fibroblast versus stroma cancer cell might not be as straightforward as described in the plan. Telling cancer cells from their healthy counterparts is generally not a trivial task (for all cancer entities), but particularly difficult for WT due to a lack of unequivocal markers.  Since the ScPCA portal for WT also contain VISIUM data, I would suggest using this data to identify with a pathologist homogeneous areas of normal kidney epithelium, normal stroma, nephrogenic rest, cancer stroma, cancer epithelium and cancer blastema to derive from the VISIUM data reliable signature for each cell population. The resolution of H&E pictures available in the portal is unfortunately not sufficient for a pathologist to perform such annotation. Better resolution for few slides might be available?

**Comment by Jen-OMalley** [comment]
Hi @maud-p. I'm Jen, the Scientific Community Manager at the Data Lab. Thank you for sharing your proposed analysis!

Have you filled out the [contributor form](https://share.hsforms.com/1MlLtkGYSQa6j23HY_0fKaw336z0) yet? On this form, you will provide the name and email address that will be associated with the AWS account that we'll create for you. We also need this form returned to ensure you have agreed to the OpenScPCA terms and conditions and other policies. Once we receive this, our team will review your proposed analyses and get back to you with next steps within 3 business days!

In the meantime, please let us know if you have any questions about OpenScPCA. We look forward to discussing more with you soon!

**Reply by maud-p** [reply]
Dear Jen, Thank you very much!

**Reply by jaclyn-taroni** [reply]
Hi @maud-p, your AWS account has been created, and you should receive an email to complete setup. [Here are instructions for setting up AWS](https://openscpca.readthedocs.io/en/latest/software-platforms/aws/joining-aws/)!

**Reply by maud-p** [reply]
Thank you, that's working nicely :)

**Comment by Jen-OMalley** [comment]
@maud-p, I also wanted to point out that another researcher previously proposed an analysis of the same Wilms Tumor dataset (Group ID: SCPCAB0010 Project ID: SCPCP000006). We are only able to provide one award per project. If more than one person is working on the same project, the researcher who completes the analysis first would be eligible to apply for the award. You are still welcome to continue with your proposed analysis! I just wanted to make sure you're aware of this. 

- All analyses that are currently in progress are linked [on the sample table here](https://github.com/AlexsLemonade/OpenScPCA-analysis/discussions/571).
- [Here are the full grant guidelines and eligibility criteria.](https://openscpca.readthedocs.io/en/latest/grant-opportunities/) 

Please let me know if you have any questions!

**Comment by allyhawkins** [comment]
Hi @maud-p, I'm Ally, one of the Data Scientists in the Data Lab. We're looking forward to having you on board as an OpenScPCA contributor! 

Please follow the below steps to [start contributing to the project](https://openscpca.readthedocs.io/en/latest/contributing-to-analyses/):

- [Follow the technical setup instructions found in the `OpenScPCA` documentation](https://openscpca.readthedocs.io/en/latest/technical-setup/).
- [File an issue to track the initiation of your analysis module](https://openscpca.readthedocs.io/en/latest/communications-tools/github-issues/).
- [Initiate your module](https://openscpca.readthedocs.io/en/latest/contributing-to-analyses/analysis-modules/creating-a-module/) and [file a pull request](https://openscpca.readthedocs.io/en/latest/contributing-to-analyses/creating-pull-requests/).

After you have initiated your module, you will be ready to continue with the rest of the analysis that you proposed. I would recommend that you break up your work into the steps that you outlined in your initial post, where each step corresponds to an issue and at least one subsequent pull request. I've left some initial thoughts for each of the steps you mentioned below. 

- Dimensionality reduction and clustering. Because you will be starting with the processed objects on the Portal, dimensionality reduction using both PCA and UMAP will already be present.  Although we do provide cluster assignments in these objects, I would recommend re-doing your own clustering. We used default parameters to apply the same clustering approach to all samples on the Portal, so I don't expect those to be the most reliable clusters. 
- Identify immune and normal cells using annotation transfer. When you get to this step, it will be important to consider what methods you will use to actually perform the label transfer. Do you have methods in mind that you have used in the past? 
- Identification of malignant cells using `InferCNV` or `Numbat`. Unfortunately, we cannot provide access to the BAM files and only have the summarized gene expression files available so you will have to use tools like `InferCNV` and `CopyKAT` for this step. 
	- I would recommend first setting up a script to run one of these methods and annotate the single cells and then in a separate pull request adding the cluster level annotation that you mention. 
	- Also it could be helpful to consider any known copy number alterations in WT and validate that those exist in the cells that are annotated as tumor cells. 
- Integrate CNV, label transfer and marker genes. It sounds like you plan on making a list of marker genes as part of this step. I would first create a metadata file that contains the marker genes for the cell types you mention prior to combining the annotations. 
- Integrate all 40 samples and validate annotations. I would recommend creating a script to perform the integration and clustering and then have a second step that actually evaluates the output and the annotations in a notebook. I'm not entirely sure what you mean by "comparing compatible cluster marker expression across all integrated datasets", but we can always discuss more as you get to this step. 
- Identification of marker genes. I think this would be a great added bonus. It might also be helpful to compare the identified marker genes to the list of marker genes that you curated in step 5. 

As for your note about using the VISIUM data, I would definitely encourage you to use that if you think it will be beneficial in annotating normal vs. malignant cell types. As far as getting access to better resolution images, you would have to reach out to the initial submitters as we only have access to the images that are on the Portal. You can find their contact information on the [project page](https://scpca.alexslemonade.org/projects/SCPCP000006). 

For more information on contributing to the project, I recommend you review these sections of the documentation: 

- [More on analysis modules](https://openscpca.readthedocs.io/en/latest/contributing-to-analyses/analysis-modules/)
- [Working with Git](https://openscpca.readthedocs.io/en/latest/contributing-to-analyses/working-with-git/)
- [Scoping and creating pull requests](https://openscpca.readthedocs.io/en/latest/contributing-to-analyses/creating-pull-requests/)

Please let me know if you have any additional questions and we are looking forward to working with you!

**Reply by maud-p** [reply]
Dear Ally, 

thank you very much for your detailed comment and very useful advice. I’ll go through the first steps, thank you for the documentation!

Regarding your comments:

•	Dimensionality reduction and clustering: good to know thanks, I’ll re-perform the clustering. Any dimensionality reduction preferred (PCA, UMAP) or can I go with the one that seems to fit the best (maybe Harmony?)

•	Label transfer: I previously used the wrapper RunAzimuth that is based on finding anchors between the query and reference. I have seen that you are mostly performing cell type annotation with SingleR which should give quite comparable results when using the same reference dataset (?). I never used it but I am happy to give a try! In both cases, I think that the best reference is the fetal kidney atlas.

•	Identification of malignant cells using InferCNV: good ideas, thanks! There are indeed some known large chromosome gain/loss that should be identified by InferCNV, I’ll check this more carefully.

•	Integrate CNV, label transfer and marker genes: to be honest, I am not sure how to use the marker genes for the annotations und might only use it as validation. I need to see once I am at this step, might be also useful to discuss more once I am in this step. But in any case, a metadata table with info and reference for each gene is definitely useful!

Maybe one question regarding the organization of the work. I think some step might be easily automated and I plan to “render” RMarkdown reports for each of the samples, is that OK? The final annotation using integration of the previous steps might however require some adjustments for each sample, not sure if we can fully automate this part, is that also OK to have different reports for this step?

Thank you very much for your help, I am really glad to be on board!

**Reply by allyhawkins** [reply]
> Dimensionality reduction and clustering: good to know thanks, I’ll re-perform the clustering. Any dimensionality reduction preferred (PCA, UMAP) or can I go with the one that seems to fit the best (maybe Harmony?)

This depends on what you are using the dimensionality reduction for. We have already calculated PCA and UMAP for you to use. 
- To perform clustering, you should be using the PCA coordinates as input. 
- To visualize the data, we recommend using the UMAP coordinates. 
- Harmony would only be if you were performing batch correction across multiple datasets. But you should start by looking at individual samples so I don't think you would be using Harmony just yet. 

>  Label transfer: I previously used the wrapper RunAzimuth that is based on finding anchors between the query and reference. I have seen that you are mostly performing cell type annotation with SingleR which should give quite comparable results when using the same reference dataset (?). I never used it but I am happy to give a try! In both cases, I think that the best reference is the fetal kidney atlas.

If you are specifically trying to do label transfer from the fetal kidney atlas, then I would use the `RunAzimuth()` function that you mention. The caveat with that function is that will be very computationally intensive as you will have to integrate each sample with the atlas, which I imagine is not a small object. 

Alternatively, you could definitely use the atlas you have as a reference with `SingleR`. `SingleR` works by first finding a set of important marker genes for each cell class in the reference. It then calculates the correlation of marker gene expression between the query and the reference for every cell and every label, assigning a score for each correlation. Ultimately, the cell label that has the highest score is assigned as the cell type. This is a much faster approach then running `RunAzimuth()` and in our hands works well when you have a reference that you feel confident about. 

> Integrate CNV, label transfer and marker genes: to be honest, I am not sure how to use the marker genes for the annotations und might only use it as validation. I need to see once I am at this step, might be also useful to discuss more once I am in this step. But in any case, a metadata table with info and reference for each gene is definitely useful!

I think using the marker gene table as validation is really helpful! Ideally, cells that are assigned to a specific cell type show higher expression of the expected marker genes over all other cells. 

> Maybe one question regarding the organization of the work. I think some step might be easily automated and I plan to “render” RMarkdown reports for each of the samples, is that OK? The final annotation using integration of the previous steps might however require some adjustments for each sample, not sure if we can fully automate this part, is that also OK to have different reports for this step?

Having a markdown report for each sample is definitely okay! And it is also okay to have reports at different steps too. Based on what you've suggested, I might take the following approach: 

 - Compile a metadata file of marker genes for expected cell types (you will be able to use this for validation at a later step)
 - Create a script to perform clustering. Here you might want to actually cluster cells across a set of parameters and store all the results. You can then evaluate the findings in the notebook mentioned below to identify the most optimal clusters. 
- Create a script that can run either `runAzimuth` or `SingleR` (or maybe one script for each) on a single sample
- Run those scripts on a handful of samples and explore the output in an exploratory notebook. You could use the marker gene list here to help validate assignments. Here I would look at each sample in it's own notebook. 

From there, you could build on your analysis (add in a script for `InferCNV` for example) and add a template markdown file that can be rendered across all samples. This will make it so that you can produce some of the same plots across every sample. Once you move to integrating everything, then I think you will need to be a little bit more manual and we can talk about organization once you get there!
